openapi.yaml

```diff
@@ -171,6 +547,365 @@ components:
           schema:
             $ref: '#/components/schemas/ErrorResult'
   schemas:
+    BookId:
+      type: string
+      description: Book ID
+      example: '26'
+    BookType:
+      type: string
+      enum:
+        - book
+    BookAvailable:
+      type: string
+      nullable: false
+      description: Availability of book (true/false).
+      example: true
+    MemberId:
+      type: string
+      description: Member ID
+      example: '001'
+    MemberType:
+      type: string
+      enum:
+        - member
+    MembershipStatus:
+      type: string
+      nullable: false
+      description: Membership status of the member (active/suspended/expired).
+      example: active
+    BorrowId:
+      type: string
+      description: Borrow ID
+      example: 'bo123'
+    BorrowType:
+      type: string
+      enum:
+        - borrow
+    BorrowReturnDate:
+      type: string
+      format: date
+      nullable: true
+      description: Returned date of the book
+      example: null
+    BorrowStatus:
+      type: string
+      nullable: true
+      description: Current borrow status of the book (ongoing/returned/overdue)
+      example: ongoing
+    BookPostBody:
+      type: object
+      required:
+        - data
+      additionalProperties: false
+      properties:
+        data:
+          type: object
+          required:
+            - type
+            - attributes
+          additionalProperties: false
+          properties:
+            type:
+              $ref: '#/components/schemas/BookType'
+            attributes:
+              allOf:
+                - $ref: '#/components/schemas/BookPostRequired'
+                - $ref: '#/components/schemas/BookPostAttributes'
+    BookPostRequired:
+      type: object
+      required:
+        - title
+        - author
+        - publicationYear
+        - genre
+    BookPostAttributes:
+      type: object
+      additionalProperties: false
+      properties:
+        title:
+          type: string
+          nullable: false
+          description: Title of book
+          example: The Alchemist
+        author:
+          type: string
+          nullable: false
+          description: Author of book
+          example: Paulo Coelho
+        publicationYear:
+          type: integer
```

Comment:
It's a string. For example 0001 != 1 in date.

Reply:
Here, It would just be a year, and it would be  basically 4 digits together as 1988, making it an integer/number. should I change it to string?

Reply:
changed to string tho

Reply:
Yes, it should be a string.